test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegratedTests.cs

```diff
@@ -695,386 +696,441 @@ public async Task InstallMultiplePackage()
-                    sourceRepositoryProvider.GetRepositories(),
-                    sourceRepositoryProvider.GetRepositories(),
-                    CancellationToken.None);
+            await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(
+                buildIntegratedProject,
+                actions,
+                new TestNuGetProjectContext(),
+                NullSourceCacheContext.Instance,
+                token);
 
-                await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(
-                    buildIntegratedProject,
-                    actions,
-                    new TestNuGetProjectContext(),
-                    NullSourceCacheContext.Instance,
-                    CancellationToken.None);
 
-                var installedPackages = await buildIntegratedProject.GetInstalledPackagesAsync(CancellationToken.None);
-                var lockFile = ProjectJsonPathUtilities.GetLockFilePath(buildIntegratedProject.JsonConfigPath);
+            // Assert
+            var installedPackages = (await buildIntegratedProject.GetInstalledPackagesAsync(token)).ToList();
+            string assetsFile = await buildIntegratedProject.GetAssetsFilePathAsync();
+            File.Exists(assetsFile).Should().BeTrue();
 
-                // Assert
-                Assert.Equal(1, actions.Count());
-                Assert.True(actions.First() is BuildIntegratedProjectAction);
-                Assert.Equal(1, installedPackages.Count());
-                Assert.True(installedPackages.Single().PackageIdentity.Version > oldJson.Version);
-                Assert.True(File.Exists(lockFile));
-            }
+            installedPackages.Count.Should().Be(1);
+            installedPackages[0].PackageIdentity.Version
+                .Should().BeGreaterThan(originalInstalledPackage.Version);
+
+            actions.Count.Should().Be(1);
+            actions[0].Should().BeOfType<BuildIntegratedProjectAction>();
         }
 
-        [Fact(Skip = "https://github.com/NuGet/Home/issues/10212")]
+        [Fact]
         public async Task UpdateMultipleAndRollback()
         {
             // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            using var testSolutionManager = new TestSolutionManager(pathContext);
+
             // This package is not compatible with netcore50 and will cause the rollback.
-            var oldVersioning = new PackageIdentity("NuGet.Versioning", NuGetVersion.Parse("1.0.5"));
+            var packageNotCompatible = new PackageIdentity("NuGet.Versioning", NuGetVersion.Parse("1.0.5"));
 
             // This package is compatible.
-            var oldJson = new PackageIdentity("Newtonsoft.Json", NuGetVersion.Parse("6.0.8"));
+            var originalPackage = new PackageIdentity("Newtonsoft.Json", NuGetVersion.Parse("6.0.8"));
 
             var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateV2OnlySourceRepositoryProvider();
```

Comment:
Create a SourceProvider based on the current source.
```suggestion
            var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(new PackageSource(pathContext.PackageSource));

```